mode_blueprints/run_report.py

```diff
@@ -0,0 +1,102 @@
+import argparse
+import sys
+import requests
+from requests.auth import HTTPBasicAuth
+import shipyard_utils as shipyard
+try:
+    import errors
+except BaseException:
+    from . import errors
+
+# create Artifacts folder paths
+base_folder_name = shipyard.logs.determine_base_artifact_folder(
+    'mode')
+artifact_subfolder_paths = shipyard.logs.determine_artifact_subfolders(
+    base_folder_name)
+shipyard.logs.create_artifacts_folders(artifact_subfolder_paths)
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--account-id', dest='account_id', required=True)
+    parser.add_argument('--report-id', dest='report_id', required=True)
+    parser.add_argument('--token-id', dest='token_id', required=True)
+    parser.add_argument('--token-password', dest='token_password', required=True)
+    args = parser.parse_args()
+    return args
+
+
+def execute_run_report(account_id, report_id, token_id, token_password):
+    """Executes a mode report run
+    see: https://mode.com/developer/api-reference/analytics/report-runs/#runReport
+    """
+    mode_api_base = f"https://app.mode.com/api/{account_id}"
+    run_report_endpoint = mode_api_base + f"/reports/{report_id}/runs"
+    parameters = {}
+
+    headers = {
+      'Content-Type': 'application/json',
+      'Accept': 'application/hal+json'
+    }
+    report_request = requests.post(run_report_endpoint,
+                          data=parameters, 
+                          headers=headers,
+                          auth=HTTPBasicAuth(token_id, token_password))
+    status_code = report_request.status_code
+    
+    # save report data
+    run_report_data = report_request.json()
+
+    
+    run_report_file_name = shipyard.files.combine_folder_and_file_name(
+        artifact_subfolder_paths['responses'],
+        f'sync_run_{report_id}_response.json')
+    shipyard.files.write_json_to_file(run_report_data, run_report_file_name)
+    
+    # handle reponse codes
+    if status_code == 202: # Report run successful
+        print("Run report for ID: {report_id} successful")
```

Comment:
Can we adjust to `Run report for ID: {report_id} was successfully triggered.` just to clarify a bit more that the run was triggered, but that doesn't mean it was successful.